tests/platform_tests/sfp/test_sfputil.py

```diff
@@ -122,54 +326,46 @@ def test_check_sfputil_eeprom(duthosts, enum_rand_one_per_hwsku_frontend_hostnam
-    tested_physical_ports = set()
-    for intf in dev_conn:
-        if intf not in xcvr_skip_list[duthost.hostname]:
-            phy_intf = portmap[intf][0]
-            if phy_intf in tested_physical_ports:
-                logging.info(
-                    "skip tested SFPs {} to avoid repeating operating physical interface {}".format(intf, phy_intf))
-                continue
-            tested_physical_ports.add(phy_intf)
-            logging.info("resetting {} physical interface {}".format(intf, phy_intf))
-            reset_result = duthost.command("{} {}".format(cmd_sfp_reset, intf))
-            assert reset_result["rc"] == 0, "'{} {}' failed".format(cmd_sfp_reset, intf)
-            time.sleep(5)
-    sleep_time = 60
-    if duthost.shell("show interfaces transceiver eeprom | grep 400ZR", module_ignore_errors=True)['rc'] == 0:
-        sleep_time = 90
-
-    logging.info("Wait some time for SFP to fully recover after reset")
-    time.sleep(sleep_time)
+    intfs_to_test_per_asic, all_intfs_per_asic = get_intfs_to_test(duthost,
+                                                                   conn_graph_facts,
+                                                                   enum_frontend_asic_index,
+                                                                   xcvr_skip_list)
 
-    logging.info("Check sfp presence again after reset")
-    sfp_presence = duthost.command(cmd_sfp_presence, module_ignore_errors=True)
+    for intf, phy_intf in intfs_to_test_per_asic.items():
+        with DisablePort(duthost, enum_frontend_asic_index, intf, phy_intf):
+            cmd_sfp_presence_per_intf = cmd_sfp_presence + " -p {}".format(intf)
 
-    # For vs testbed, we will get expected Error code `ERROR_CHASSIS_LOAD = 2` here.
-    if duthost.facts["asic_type"] == "vs" and sfp_presence['rc'] == 2:
-        pass
-    else:
-        assert sfp_presence['rc'] == 0, "Run command '{}' failed".format(cmd_sfp_presence)
-
-    parsed_presence = parse_output(sfp_presence["stdout_lines"][2:])
-    for intf in dev_conn:
-        if intf not in xcvr_skip_list[duthost.hostname]:
-            assert intf in parsed_presence, "Interface is not in output of '{}'".format(cmd_sfp_presence)
-            assert parsed_presence[intf] == "Present", "Interface presence is not 'Present'"
-
-    logging.info("Check interface status")
-    mg_facts = duthost.get_extended_minigraph_facts(tbinfo)
-    intf_facts = duthost.interface_facts(up_ports=mg_facts["minigraph_ports"])["ansible_facts"]
-    assert len(intf_facts["ansible_interface_link_down_ports"]) == 0, \
-        "Some interfaces are down: {}".format(intf_facts["ansible_interface_link_down_ports"])
+            cmd_sfp_reset_intf = "{} {}".format(cmd_sfp_reset, intf)
+            logging.info("resetting {} physical interface {}".format(intf, phy_intf))
+            reset_result = duthost.command(cmd_sfp_reset_intf)
```

Comment:
@longhuan-cisco In case of breakout ports, shouldn't we issue reset on one of the subports in breakout group and issue shut/no shut for other subports in the corresponding breakout group?

Reply:
agree, updated as the same. thanks